valik kod/lab2.cpp

```diff
@@ -0,0 +1,161 @@
+#include <iostream>
+#include <initializer_list>
+#include <functional>
+#include <memory>
+#include <stdexcept>
+
+template <typename T>
+class LinkedList
+{
+private:
+    struct Node
+    {
+        T data;
+        std::unique_ptr<Node> next;
+        Node(const T &data) : data(data), next(nullptr) {}
+    };
+
+    std::unique_ptr<Node> head;
+    size_t size_;
+
```

Comment:
Code Structure: Ensured consistency in method naming and code style.

Reply:
ok